packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -36,6 +37,7 @@ export interface ExtensionSpecification<TConfiguration extends BaseConfigType =
     apiKey: string,
     moduleId?: string,
   ) => Promise<{[key: string]: unknown} | undefined>
+  getConfigurationObject?: (config: TConfiguration) => {[key: string]: unknown}
```

Comment:
we are trying to simplify the extension specification definition, so I would try to avoid adding this kind of new specific implementations.
In this case, I would use the new `appModuleFeatures` added to the spec to implement this behaviour when you need it. 

For example, instead of:
```ts
const customConfigurationObject =
      specification.getConfigurationObject?.(configurationObject as BaseConfigType) || configurationObject
```

we could use something similar to this:
```ts
    let customConfigurationObject = configurationObject as object
    if (specification.appModuleFeatures().includes('app_config')) {
      customConfigurationObject = {
        ...customConfigurationObject,
        type: specification.identifier,
        name: specification.externalName,
      }
    }
```

Reply:
Ooh ok. Will do